### PR TITLE
Add Conn.SetAuthorizer method

### PR DESCRIPTION
**Pull request by zombiezen**
Implementation note: this avoids the cgo integer/pointer conversion issue in #87 by performing the conversion entirely in C. This might be helpful as a jumping off point.

### CODE REVIEW

**Comment by AdamSLevy** [comment]
Many thanks for the contribution! I'll review this later this week.

**Comment by zombiezen** [comment]
Hey @AdamSLevy, mind taking a look at this PR and #89?

**Comment by AdamSLevy** [comment]
@zombiezen Sorry for the delay on this. Started a new job recently and I've been swamped. I'll try to review this week. The intentions look good.

**Comment by zombiezen** [comment]
No worries, same here. Thank you for maintaining this package: it has helped me do a lot of cool stuff!

File: auth.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	Authorize(action OpType, info ActionInfo) AuthResult
+}
+
+// ActionInfo holds information about an action to be authorized.
+type ActionInfo struct {
+	Arg1     string
+	Arg2     string
+	Database string
+	Trigger  string
+}
```
Comment (AdamSLevy) [inline auth.go:30]: I'm going to make this type more specific and useful to Go devs by removing Arg1 and Arg2 and replacing them with name specific types that match the SQLite documentation such as Index, Table. View, and then having the trampoline populate the relevant values based on the action.

File: auth.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import "C"
+import (
+	"errors"
+	"sync"
+)
+
+// An Authorizer is called during statement preparation to see whether an action
+// is allowed by the application. See https://sqlite.org/c3ref/set_authorizer.html
+type Authorizer interface {
+	Authorize(action OpType, info ActionInfo) AuthResult
```
Comment (AdamSLevy) [inline auth.go:21]: Going to move action OpType into ActionInfo.

File: auth.go (hunk 1 above)
Comment (AdamSLevy) [reply auth.go:30]: As such it makes sense to merge the OpType var to this struct.

**Pull request merged**